### PR TITLE
feat: Support multiple rules from notifications in UI

### PR DESCRIPTION
**Pull request by ptbrowne**
- Extracted code from category budgets to add support for multiple rules
  in BalanceLower and TransactionGreater
- Multiple card for multiple rules
- Add rule Button
- Remove rule Button inside modal if more than 1 alert

The Rules component is responsible of

- displaying a list of EditableSettingCard
- merging the edit of an object from a card inside the list
- displaying a create button

### CODE REVIEW

File: test/e2e/alerts.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  account: ISA_CHECKING_ACCOUNT_ID,
+  amount: 0,
+  label: 'Fake transaction'
+}
+
+const scenarios = {
+  balanceLower1: {
+    description: 'Notification for Louise checking account',
+    expectedEmail: {
+      subject: "Balance_alert:_'Louise_checkings'_account_is_at_100€"
```
Comment (drazik) [inline test/e2e/alerts.js]: Why do we have underscores in the mail subject?

Comment (ptbrowne) [reply test/e2e/alerts.js]: Mailhog returns the subject of emails with underscores. Apparently the rfc2047 stipulates that in email headers, underscores should interpreted as spaces. I will fix this by passing the subject into a replacer. Thanks for noticing, I was going to sweep this under the carpet.

File: src/ducks/settings/BalanceLowerRules.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    onChangeRules,
+    t
+  } = props
+
+  const initialRules = ensureNewRuleFormat(rawInitialRules)
+  const onError = () =>
+    Alerter.error(t('Settings.budget-category-alerts.saving-error'))
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply src/ducks/settings/BalanceLowerRules.jsx]: Removed in a later refactor, thanks !

File: src/components/EditionModal/EditionModal.jsx [hunk 3]
```diff
@@ -73,46 +74,49 @@ const ChoosingSwitch = ({ choosing }) => {
...
+    fieldLabels,
+    onRequestChooseField,
+    onChangeField,
+    t
+  }) => {
+    return (
+      <ModalSections>
+        {fieldOrder.map(fieldName => {
+          const FieldSection = SectionsPerType[fieldSpecs[fieldName].type]
+          const fieldSpec = fieldSpecs[fieldName]
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply src/components/EditionModal/EditionModal.jsx]: Indeed :) Thanks

**Comment by ptbrowne** [comment]
Flo m'a demandé des marges, j'en ai profité pour rationnaliser encore plus les histoires de ToggleRow/Wrapper et Sections. Il n'y a presque plus rien dans ToggleRow et on utilise + les Stack :) Voir https://github.com/cozy/cozy-banks/pull/1726/commits/c2e95a6b5f6d7bc07813a5fed502971a640cdeeb

File: src/ducks/settings/Rules.styl [hunk 4]
```diff
@@ -0,0 +1,2 @@
+.AddRuleButton:first-child:last-child
```
Comment (drazik) [inline src/ducks/settings/Rules.styl]: I think you can use [`:only-child`](https://developer.mozilla.org/fr/docs/Web/CSS/:only-child)

Comment (ptbrowne) [reply src/ducks/settings/Rules.styl]: TIL

**Pull request merged**